### PR TITLE
Add setting to hide Calva's REPL UI

### LINKED ISSUE #1005: Calva setting to use only static parts
It was an idea discussed with @PEZ [here](https://clojurians.slack.com/archives/CPABC1H61/p1612181210149200), where may have users that like Calva features like `clojure-lsp` integration but don't like the REPL or prefer another tool as the REPL.
Calva would reach those users if it could have a setting where it'd disable the REPL feature/keybindings etc (Probably in the future, extract it to other dep/extension?)

Just an idea though, I don't think it should be a priority too ;)

### PR DESCRIPTION
**Pull request by PEZ**
## What has Changed?

Following discussions in the `#lsp` and `#chlorine-clover` channels at Clojurians Slack, regarding how to give `clojure-lsp` and things like Calva's formatter, Paredit, code highlighting and such to Clover users, @ericdallo suggested a not-so-costly alternative: Since Calva brings all this. _Make it easier to use Calva only for its static services._

Which makes a lot of sense. I have spent a lot of time trying to  make Calva format etcetera available seperately, to pull apart what I once joined (for good reasons). I still think that is the real solution, but it is hard for me to motivate myself using time I can spend on serving Calva users, to make non-Calva-users happy. This is a good interim solution, until someone signs up for the real one. It also happens to possibly attract more Calva users. Haha. 😄 (Jokes, aside, it is good that more people have the Calva REPL a keystroke away, we do think it is quite awesome, right?.)

In this PR I have added a user setting, `calva.hideREPLUi` that affect how the status bar, command palette and editor context menus behave. When the settings is active, the status bar won't show Calva REPL items, the command palette won't show Calva REPL commands and the editor context menu won't either. The only commands still visible are for connecting or starting a REPL. And when the user does that, the REPL UI is enabled again, until Calva's REPL is disconnected.

Fixes #1005 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. NB: *There is a CircleCI bug that makes the Artifacts hard to find. Please see [this issue](https://discuss.circleci.com/t/artifacts-tab-not-showing-unless-logged-in/32433) for workarounds.*
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Added to or updated docs in this branch, if appropriate

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [ ] Made sure the PR is directed at the `dev` branch (unless reasons).
- [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and tested it there if so.
- [ ] Read the source changes.
- [ ] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] If need be, had a chat within the team about particular changes.

Ping @pez, @kstehn, @cfehse, @bpringe

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -501,6 +501,11 @@
                         "description": "Activate keybindings.",
                         "default": true,
                         "scope": "window"
+                    },
+                    "calva.hideREPLUi": {
```
Comment (bpringe) [inline package.json]: Purely a style thing here, but I vote for this to be cased as `calva.hideReplUi` :smiley:

Comment (PEZ) [reply package.json]: I'll consider it. There was some other setting that used this style, so I chose to follow that. But it does hurt the eye a bit.

Comment (ericdallo) [reply package.json]: That's why I prefer `kebab-case` 😛 😂

Comment (PEZ) [reply package.json]: kebab-case-ftw

**Pull request merged**